# Guided Project: Analyzing CIA Factbook Data Using SQL

The `US Central Intelligence Agency` publishes the <a href='https://www.cia.gov/the-world-factbook/'>CIA factbook</a>, usually referred to as the World Factbook, once a year. It gives fundamental insight by condensing data about all nations and locations of the world. The factbook includes a variety of regional and demographic information.

![Image of Yaktocat](https://miro.medium.com/max/720/1*QPDFLifvVMnzrIrx80e_bw.webp)


The Factbook contains demographic information like the following:

- Population — the global population.
- Population_growth — the `annual population growth rate`, as a percentage.
- Area — the `total land` and `water area`.

In order to examine data from this <a href='https://dsserver-prod-resources-1.s3.amazonaws.com/257/factbook.db'>SQLite factbook.db</a> database for this project, we will use Jupyter Notebook and SQL. Based on demographic variables including population, mortality, and fertility, the aim is to determine which countries and regions are leading and trailing in the world.

## Connecting to the Database

In [2]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

# Overview of the Data

Throughout this course, we introduced the name of the table and what the table looked like in the jobs.db database.

In [3]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

#### Notes
We have two tables in the database, sqlite_sequence and facts.
- The are no columns in the sqlite_sequence table.
- The facts table contains information on population, birth rate, migration rate and many more: which is what is needed.

We will work with the facts table henceforth.

In [6]:
%%sql
-- Displaying the first five rows in facts table
SELECT *
  FROM facts
LIMIT 5

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

# Summary Statistics 

Here are the descriptions for some of the columns:
    The facts table comprises 11 columns with intuitive names:

- name — the name of the country.
- area— the country's total area (both land and water).
- area_land — the country's land area in <a href='https://www.cia.gov/library/publications/the-world-factbook/rankorder/2147rank.html'>square kilometers.</a>
- area_water — the country's waterarea in <a href='https://www.cia.gov/library/publications/the-world-factbook/rankorder/2147rank.html'>square kilometers.</a>
- population — the country's population.
- population_growth— the country's population growth as a percentage.
- birth_rate — the country's birth rate, or the number of births per year per 1,000 people.
- death_rate — the country's death rate, or the number of death per year per 1,000 people.

#### Let's check the total number of records in the facts table:

In [7]:
%%sql
-- Total number of records
SELECT COUNT(*) AS num_records,
       COUNT(DISTINCT name) AS unique_countries
  FROM facts;

Done.


[(261, 261)]

We have 261 records and countries, which seems impossible! Since we know that there are only 195 countries in the world right now. The CIA's classification system for information holds the key to the reason why we have more country. The 267 entities that make up The World Factbook as of July 2011 fall into the following categories:

- Independent countries: These are individuals who have been "politically organised into a sovereign state with a defined territory," according to the CIA. This category includes 195 different entities.
- Others: Regions excluded from the list of independent nations. There are currently only two: Taiwan and the EU.
- Dependencies and Areas of Special Sovereignty: Regions connected to another nation. Total, there are 58.
- Miscellaneous: Antarctica and places in dispute. There are six of these categories.
- Other entities: Oceans and the world. The world and its five oceans (the World entry is intended as a summary of the other entries).

    
Additional information about this classification can be found in the <a href='https://en.wikipedia.org/wiki/The_World_Factbook#:~:text=Entities%20listed,-Main%20article%3A%20List&text=The%20CIA%20defines%20these%20as,category%2C%20there%20are%20195%20entities.&text=Places%20set%20apart%20from%20the,Taiwan%20and%20the%20European%20Union.'>World factbook wikipedia page.</a>

We can examine our data for these occurrences:

In [9]:
%%sql
-- check for some instances of "non-country" data
SELECT *
  FROM facts
 WHERE name LIKE '%union' 
    OR name LIKE '%ocean'
    OR name LIKE '%island'
    OR name = 'World';

Done.


[(197, 'ee', 'European Union', 4324782, None, None, 513949445, 0.25, 10.2, 10.2, 2.5),
 (199, 'kt', 'Christmas Island', 135, 135, 0, 1530, 1.11, None, None, None),
 (203, 'nf', 'Norfolk Island', 36, 36, 0, 2210, 0.01, None, None, None),
 (208, 'ip', 'Clipperton Island', 6, 6, 0, None, None, None, None, None),
 (222, 'bv', 'Bouvet Island', 49, 49, 0, None, None, None, None, None),
 (244, 'bq', 'Navassa Island', 5, 5, 0, None, None, None, None, None),
 (248, 'wq', 'Wake Island', 6, 6, 0, None, None, None, None, None),
 (256, 'xq', 'Arctic Ocean', None, None, None, None, None, None, None, None),
 (257, 'zh', 'Atlantic Ocean', None, None, None, None, None, None, None, None),
 (258, 'xo', 'Indian Ocean', None, None, None, None, None, None, None, None),
 (259, 'zn', 'Pacific Ocean', None, None, None, None, None, None, None, None),
 (260, 'oo', 'Southern Ocean', None, None, None, None, None, None, None, None),
 (261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

We should take care to prevent these occurrences from having an impact on the analysis process because they do occur.

Now that we have a basic understanding of the data and any hazards to watch out for, we can move on to analysis.

#### Examining the Total Population

Let's start by calculating some summary statistics and look for any outlier countries.

In [12]:
%%sql
SELECT MIN(population) AS min_pop,
       MAX(population) AS max_pop,
       MIN(population_growth) AS min_pop_growth,
       CAST(AVG(population) AS Integer) AS avg_population,
       MAX(population_growth) max_pop_growth 
  FROM facts;

Done.


[(0, 7256490011, 0.0, 62094928, 4.02)]

Surprising! Some nations and other entities are devoid of people. Over 7 billion people are found in another entity. Given that there are already <a href='https://www.worldometers.info/world-population'>7.96 billion</a> people living in the earth, this population appears too good to be true (Worldometer Aug, 2022). The best course of action is to define these entities:

In [15]:
%%sql 
-- Investigate the unusual countries/entities
SELECT name, population
  FROM facts
 WHERE population = (SELECT MIN(population) FROM facts) 
    OR population = (SELECT MAX(population) FROM facts);

Done.


[('Antarctica', 0), ('World', 7256490011)]

The existence of items like World and Antarctica explains the peculiar population values, correlating with past observations. Antarctica has no native people, only research employees who periodically visit its numerous research stations, according to the World population review.

The value of the world population is another interesting point to make (7,256,490,011). According to a simple review of global population trends, the CIA could have gathered the data between 2014 and 2015.

The population of the world around 2014 and 2015 is around 7,379,797,139/7,295,290,765 according to <a href=' https://www.worldometers.info/world-population/world-population-by-year/'> Worldometers</a>	

# Exploring Outliers

We see a few interesting things in the summary statistics:

<ul>
    <li>There's a country with a population of 0</li>
    <li>There's a country with a population of 7256490011 (or more than 7.2 billion people)</li>
</ul>
 
 Let's use subqueries to zoom in on just these countries without using the specific values.

In [16]:
%%sql
SELECT *
  FROM facts
 WHERE population == (SELECT MIN(population)
                        FROM facts
                     );

Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

It seems like the table contains a row for the whole world, which explains the population of over 7.2 billion. It also seems like the table contains a row for Antarctica, which explains the population of 0. This seems to match the CIA Factbook page for Antarctica:

In [17]:
%%sql
SELECT *
  FROM facts
 WHERE population == (SELECT MAX(population)
                        FROM facts
                     );

Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

Now that we know this, we should recalculate the summary statistics we calculated earlier — this time excluding the row for the whole world.

# Exploring Average Population and Area

In [12]:
%%sql
SELECT MIN(population) AS min_pop,
       MAX(population) AS max_pop,
       MIN(population_growth) AS min_pop_growth,
       MAX(population_growth) AS max_pop_growth 
  FROM facts
 WHERE name <> 'World';

Done.


[(0, 1367485388, 0.0, 4.02)]

The average population is 32 million people, with a minimum population of 48 and a high population of approximately 1.3 billion. The most and least populous nations or regions can now be determined.


In [15]:
%%sql 
SELECT AVG(population) as avg_pol,
       AVG(area) as avg_area
    FROM facts

Done.


[(62094928.32231405, 555093.546184739)]

To finish, we'll build on the query we wrote for the previously to find countries that are densely populated. We'll identify countries that have the following:

- Above-average values for population.
- Below-average values for area.

In [16]:
%%sql
SELECT *
  FROM facts
 WHERE population > (SELECT AVG(population)
                        FROM facts
                     WHERE name <> 'World'
                     );

Done.


[(14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (24, 'br', 'Brazil', 8515770, 8358140, 157630, 204259812, 0.77, 14.46, 6.58, 0.14),
 (37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44),
 (40, 'cg', 'Congo, Democratic Republic of the', 2344858, 2267048, 77810, 79375136, 2.45, 34.88, 10.07, 0.27),
 (53, 'eg', 'Egypt', 1001450, 995450, 6000, 88487396, 1.79, 22.9, 4.77, 0.19),
 (58, 'et', 'Ethiopia', 1104300, None, 104300, 99465819, 2.89, 37.27, 8.19, 0.22),
 (61, 'fr', 'France', 643801, 640427, 3374, 66553766, 0.43, 12.38, 9.16, 1.09),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (77, 'in', 'India', 3287263, 2973193, 314070, 1251695584, 1.22, 19.55, 7.32, 0.04),
 (78, 'id', 'Indonesia', 1904569, 1811569, 93000, 255993674, 0.92, 16.72, 6.37, 1.16),
 (79, 'ir', 'Iran', 1648195, 1531595, 116600, 81824270, 1.2, 17.99, 5.94, 0.07),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (114, 'mx', 'Mexico', 1964375, 1943945, 20430, 121736809, 1.18, 18.78, 5.26, 1.68),
 (129, 'ni', 'Nigeria', 923768, 910768, 13000, 181562056, 2.45, 37.64, 12.9, 0.22),
 (132, 'pk', 'Pakistan', 796095, 770875, 25220, 199085847, 1.46, 22.58, 6.49, 1.54),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09),
 (143, 'rs', 'Russia', 17098242, 16377742, 720500, 142423773, 0.04, 11.6, 13.69, 1.69),
 (173, 'th', 'Thailand', 513120, 510890, 2230, 67976405, 0.34, 11.19, 7.8, 0.0),
 (179, 'tu', 'Turkey', 783562, 769632, 13930, 79414269, 1.26, 16.33, 5.88, 2.16),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54),
 (186, 'us', 'United States', 9826675, 9161966, 664709, 321368864, 0.78, 12.49, 8.15, 3.86),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3),
 (197, 'ee', 'European Union', 4324782, None, None, 513949445, 0.25, 10.2, 10.2, 2.5),
 (261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

In [18]:
%%sql
SELECT *
  FROM facts
 WHERE area < (SELECT avg(area)
                        FROM facts
                       WHERE name <> 'World'
                     );

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46),
 (7, 'ar', 'Argentina', 2780400, 2736690, 43710, 43431886, 0.93, 16.64, 7.33, 0.0),
 (9, 'as', 'Australia', 7741220, 7682300, 58920, 22751014, 1.07, 12.15, 7.14, 5.65),
 (21, 'bl', 'Bolivia', 1098581, 1083301, 15280, 10800882, 1.56, 22.76, 6.52, 0.62),
 (23, 'bc', 'Botswana', 581730, 566730, 15000, 2182719, 1.21, 20.96, 13.39, 4.56),
 (24, 'br', 'Brazil', 8515770, 8358140, 157630, 204259812, 0.77, 14.46, 6.58, 0.14),
 (28, 'bm', 'Burma', 676578, 653508, 23070, 56320206, 1.01, 18.39, 7.96, 0.28),
 (32, 'ca', 'Canada', 9984670, 9093507, 891163, 35099836, 0.75, 10.28, 8.42, 5.66),
 (34, 'ct', 'Central African Republic', 622984, 622984, 0, 5391539, 2.13, 35.08, 13.8, 0.0),
 (36, 'ci', 'Chile', 756102, 743812, 12290, 17508260, 0.82, 13.83, 6.0, 0.34),
 (37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44),
 (38, 'co', 'Colombia', 1138910, 1038700, 100210, 46736728, 1.04, 16.47, 5.4, 0.64),
 (40, 'cg', 'Congo, Democratic Republic of the', 2344858, 2267048, 77810, 79375136, 2.45, 34.88, 10.07, 0.27),
 (53, 'eg', 'Egypt', 1001450, 995450, 6000, 88487396, 1.79, 22.9, 4.77, 0.19),
 (58, 'et', 'Ethiopia', 1104300, None, 104300, 99465819, 2.89, 37.27, 8.19, 0.22),
 (61, 'fr', 'France', 643801, 640427, 3374, 66553766, 0.43, 12.38, 9.16, 1.09),
 (77, 'in', 'India', 3287263, 2973193, 314070, 1251695584, 1.22, 19.55, 7.32, 0.04),
 (78, 'id', 'Indonesia', 1904569, 1811569, 93000, 255993674, 0.92, 16.72, 6.37, 1.16),
 (79, 'ir', 'Iran', 1648195, 1531595, 116600, 81824270, 1.2, 17.99, 5.94, 0.07),
 (87, 'kz', 'Kazakhstan', 2724900, 2699700, 25200, 18157122, 1.14, 19.15, 8.21, 0.41),
 (88, 'ke', 'Kenya', 580367, 569140, 11227, 45925301, 1.93, 26.4, 6.89, 0.22),
 (100, 'ly', 'Libya', 1759540, 1759540, 0, 6411776, 2.23, 18.03, 3.58, 7.8),
 (105, 'ma', 'Madagascar', 587041, 581540, 5501, 23812681, 2.58, 32.61, 6.81, 0.0),
 (109, 'ml', 'Mali', 1240192, 1220190, 20002, 16955536, 2.98, 44.99, 12.89, 2.26),
 (112, 'mr', 'Mauritania', 1030700, 1030700, 0, 3596702, 2.23, 31.34, 8.2, 0.83),
 (114, 'mx', 'Mexico', 1964375, 1943945, 20430, 121736809, 1.18, 18.78, 5.26, 1.68),
 (118, 'mg', 'Mongolia', 1564116, 1553556, 10560, 2992908, 1.31, 20.25, 6.35, 0.84),
 (121, 'mz', 'Mozambique', 799380, 786380, 13000, 25303113, 2.45, 38.58, 12.1, 1.98),
 (122, 'wa', 'Namibia', 824292, 823290, 1002, 2212307, 0.59, 19.8, 13.91, 0.0),
 (129, 'ni', 'Nigeria', 923768, 910768, 13000, 181562056, 2.45, 37.64, 12.9, 0.22),
 (132, 'pk', 'Pakistan', 796095, 770875, 25220, 199085847, 1.46, 22.58, 6.49, 1.54),
 (137, 'pe', 'Peru', 1285216, 1279996, 5220, 30444999, 0.97, 18.28, 6.01, 2.53),
 (143, 'rs', 'Russia', 17098242, 16377742, 720500, 142423773, 0.04, 11.6, 13.69, 1.69),
 (151, 'sa', 'Saudi Arabia', 2149690, 2149690, 0, 27752316, 1.46, 18.51, 3.33, 0.55),
 (160, 'so', 'Somalia', 637657, 627337, 10320, 10616380, 1.83, 40.45, 13.62, 8.49),
 (161, 'sf', 'South Africa', 1219090, 1214470, 4620, 53675563, 1.33, 20.75, 9.91, 2.42),
 (162, 'od', 'South Sudan', 644329, None, None, 12042910, 4.02, 36.91, 8.18, 11.47),
 (165, 'su', 'Sudan', 1861484, None, None, 36108853, 1.72, 29.19, 7.66, 4.29),
 (172, 'tz', 'Tanzania', 947300, 885800, 61500, 51045882, 2.79, 36.39, 8.0, 0.54),
 (179, 'tu', 'Turkey', 783562, 769632, 13930, 79414269, 1.26, 16.33, 5.88, 2.16),
 (183, 'up', 'Ukraine', 603550, 579330, 24220, 44429471, 0.6, 10.72, 14.46, 2.25),
 (186, 'us', 'United States', 9826675, 9161966, 664709, 321368864, 0.78, 12.49, 8.15, 3.86),
 (191, 've', 'Venezuela', 912050, 882050, 30000, 29275460, 1.39, 19.16, 5.31, 0.0),
 (194, 'za', 'Zambia', 752618, 743398, 9220, 15066266, 2.88, 42.13, 12.67, 0.68),
 (197, 'ee', 'European Union', 4324782, None, None, 513949445, 0.25, 10.2, 10.2, 2.5),
 (207, 'gl', 'Greenland', 2166086, 2166086, 

### Which country has the most people? Which country has the highest growth rate?

#### The top 5 countries in terms of total population

In [20]:
%%sql 
SELECT name AS top_5_populated,
       population
  FROM facts
 WHERE population <> 0 
   AND name <> 'World'
 ORDER BY 2 DESC
 LIMIT 5;

Done.


[('China', 1367485388),
 ('India', 1251695584),
 ('European Union', 513949445),
 ('United States', 321368864),
 ('Indonesia', 255993674)]

The European Union should be excluded because it is not a nation. We will also determine the percentage of the world's population that is included within each nation.

In [22]:
%%sql 
SELECT name AS "5 Most Populated Countries",
       population,
       ROUND(
              CAST(population AS Float) / 
              CAST((SELECT MAX(population) FROM facts) AS Float)
             , 3) AS world_proportion
  FROM facts
 WHERE population <> 0 
   AND name NOT IN ('World', 'European Union')
 ORDER BY 2 DESC
 LIMIT 5;

Done.


[('China', 1367485388, 0.188),
 ('India', 1251695584, 0.172),
 ('United States', 321368864, 0.044),
 ('Indonesia', 255993674, 0.035),
 ('Brazil', 204259812, 0.028)]

- With 1.37 billion inhabitants, China is the most populous nation on earth. India, with a population of roughly 1.26 billion, is next. China (18.8%) and India (17.2%) together account for 36% of the world's population.

- Brazil (2.8%), Indonesia (3.5%), and the United States (4.4%) are far behind with a combined population of 11% of the entire planet.

- The top 5 nations account for 46% of the world's population overall.

In our database, there are some territories with null values of the population.

In [57]:
%%sql
SELECT name, population
  FROM facts
WHERE population IS NULL;

Done.


[('Ashmore and Cartier Islands', None),
 ('Coral Sea Islands', None),
 ('Heard Island and McDonald Islands', None),
 ('Clipperton Island', None),
 ('French Southern and Antarctic Lands', None),
 ('Bouvet Island', None),
 ('Jan Mayen', None),
 ('British Indian Ocean Territory', None),
 ('South Georgia and South Sandwich Islands', None),
 ('Navassa Island', None),
 ('Wake Island', None),
 ('United States Pacific Island Wildlife Refuges', None),
 ('Paracel Islands', None),
 ('Spratly Islands', None),
 ('Arctic Ocean', None),
 ('Atlantic Ocean', None),
 ('Indian Ocean', None),
 ('Pacific Ocean', None),
 ('Southern Ocean', None)]

All these territories represent or different uninhabited islands, or the oceans.

In [23]:
%%sql 
SELECT name AS top_5_growth_rate,
       population_growth
  FROM facts
 WHERE population <> 0 
   AND name <> 'World'
 ORDER BY 2 DESC
 LIMIT 5;

Done.


[('South Sudan', 4.02),
 ('Malawi', 3.32),
 ('Burundi', 3.28),
 ('Niger', 3.25),
 ('Uganda', 3.24)]

The rate may be positive or negative. The growth rate is a factor in determining how great a burden would be imposed on a country by the changing needs of its people for infrastructure (e.g., schools, hospitals, housing, roads), resources (e.g., food, water, electricity), and jobs. Rapid population growth can be seen as threatening by neighboring countries.

According to <a href='https://worldpopulationreview.com/countries/south-sudan-population'>Worldpopulation review </a> In the early 2000s, South Sudan took part in UN attempts to put an end to the Darfur genocide. Due to the influx of refugees into the area during this time, the population grew significantly.

## Countries with the smallest total population¶

In [24]:
%%sql 
SELECT name AS "5 Least Populated Countries",
       population
  FROM facts
 WHERE population <> 0 
   AND name <> 'Antartica'
 ORDER BY 2
 LIMIT 5;

Done.


[('Pitcairn Islands', 48),
 ('Cocos (Keeling) Islands', 596),
 ('Holy See (Vatican City)', 842),
 ('Niue', 1190),
 ('Tokelau', 1337)]

Pitcairn Islands (48 residents), Cocos Islands (596), Holy See (842), Niue (1190), and Tokelau are the least populous territory (1337).

The least populated nations are dependencies, aside from the Holy See, the global administration of the Catholic Church that is based in Vatican City. Niue and Tokelau are dependencies of New Zealand, Cocos Island is a dependence of Australia, and Pitcairn Island is a dependency of the United Kingdom. The current dependencies of the world are listed below.

### Exploring Area 

#### Which countries have the highest ratios of water to land? Which countries have more water than land?

Let's find TOP5 and BOTTOM5 countries by area

In [59]:
%%sql
SELECT name, area AS top5_area
  FROM facts
WHERE area NOT NULL
ORDER BY area DESC
LIMIT 5;

Done.


[('Russia', 17098242),
 ('Canada', 9984670),
 ('United States', 9826675),
 ('China', 9596960),
 ('Brazil', 8515770)]

Russia is the largest nation, with Canada, the United States, China, and Brazil trailing far behind.

In [61]:
%%sql
SELECT name, area AS bottom5_area
  FROM facts
WHERE area NOT NULL
ORDER BY area
LIMIT 5;

Done.


[('Holy See (Vatican City)', 0),
 ('Monaco', 2),
 ('Coral Sea Islands', 3),
 ('Ashmore and Cartier Islands', 5),
 ('Navassa Island', 5)]

The smallest country is Vatican, followed by Monaco and 3 island groups.

Next, we're going to explore what countries have the highest ratio of water to land

In [58]:
%%sql
SELECT name, area_water, area_land,
      ROUND(CAST(area_water AS Float)/ CAST(area_land AS Float), 1) AS water_to_land
    FROM facts
WHERE area_land != 0 
  AND area_water NOT NULL
ORDER BY water_to_land DESC
LIMIT 10;

Done.


[('British Indian Ocean Territory', 54340, 60, 905.7),
 ('Virgin Islands', 1564, 346, 4.5),
 ('Puerto Rico', 4921, 8870, 0.6),
 ('Bahamas, The', 3870, 10010, 0.4),
 ('Guinea-Bissau', 8005, 28120, 0.3),
 ('Malawi', 24404, 94080, 0.3),
 ('Eritrea', 16600, 101000, 0.2),
 ('Liberia', 15049, 96320, 0.2),
 ('Netherlands', 7650, 33893, 0.2),
 ('Uganda', 43938, 197100, 0.2)]

Apart from an obvious outlier British Indian Ocean Territory, this looks quite incorrect. Accroding to <a href='https://www.nationmaster.com/country-info/stats/Geography/Area/Water'>Nationmaster</a> and other <a href= 'https://www.researchgate.net/publication/271659998_Preliminary_analysis_of_spatiotemporal_pattern_of_global_land_surface_water#pf5'>researchers</a> the water ratio of Canada is the highest, and those of China and Argentina are the lowest. The value of water ratio can partially reflect the situation of water deficit for a country

## Which countries will add the most people to their populations next year?

In [65]:
%%sql 
SELECT name,
       population_growth AS top_15_growth_rate
  FROM facts
 WHERE population <> 0 
   AND name <> 'World'
 ORDER BY 2 DESC
 LIMIT 15;

Done.


[('South Sudan', 4.02),
 ('Malawi', 3.32),
 ('Burundi', 3.28),
 ('Niger', 3.25),
 ('Uganda', 3.24),
 ('Qatar', 3.07),
 ('Burkina Faso', 3.03),
 ('Mali', 2.98),
 ('Cook Islands', 2.95),
 ('Iraq', 2.93),
 ('Ethiopia', 2.89),
 ('Zambia', 2.88),
 ('Western Sahara', 2.82),
 ('Gaza Strip', 2.81),
 ('Tanzania', 2.79)]

In [66]:
%%sql
SELECT name, population_growth AS bottom15_population_growth
  FROM facts
WHERE population_growth NOT NULL
ORDER BY population_growth
LIMIT 20;

Done.


[('Holy See (Vatican City)', 0.0),
 ('Cocos (Keeling) Islands', 0.0),
 ('Greenland', 0.0),
 ('Pitcairn Islands', 0.0),
 ('Greece', 0.01),
 ('Norfolk Island', 0.01),
 ('Tokelau', 0.01),
 ('Falkland Islands (Islas Malvinas)', 0.01),
 ('Guyana', 0.02),
 ('Slovakia', 0.02),
 ('Tonga', 0.03),
 ('Niue', 0.03),
 ('Svalbard', 0.03),
 ('Russia', 0.04),
 ('Georgia', 0.08),
 ('Maldives', 0.08),
 ('Poland', 0.09),
 ('Portugal', 0.09),
 ('Andorra', 0.12),
 ('Monaco', 0.12)]

Small nations and islands—some of which we have previously seen earlier in the project—Greenland, a few nations in Western Europe (Greece, Portugal, Andorra, Monaco), certain nations in Eastern Europe (Slovakia, Poland), Russia, and Georgia are among those with the lowest to zero population growth rates. It's curious that no country saw negative population growth in 2015.

### Exploration the birth and date rate

In [63]:
%%sql
SELECT name, birth_rate AS top20_birth_rate
  FROM facts
ORDER BY birth_rate DESC
LIMIT 20;

Done.


[('Niger', 45.45),
 ('Mali', 44.99),
 ('Uganda', 43.79),
 ('Zambia', 42.13),
 ('Burkina Faso', 42.03),
 ('Burundi', 42.01),
 ('Malawi', 41.56),
 ('Somalia', 40.45),
 ('Angola', 38.78),
 ('Mozambique', 38.58),
 ('Afghanistan', 38.57),
 ('Nigeria', 37.64),
 ('Ethiopia', 37.27),
 ('Sierra Leone', 37.03),
 ('South Sudan', 36.91),
 ('Chad', 36.6),
 ('Tanzania', 36.39),
 ('Cameroon', 36.17),
 ('Benin', 36.02),
 ('Congo, Republic of the', 35.85)]

The Highest birth rate, which can reach 46 births per 1,000 people, is seen predominantly in African nations, including Afghanistan.

In [67]:
%%sql
SELECT name, birth_rate AS bottom20_birth_rate
  FROM facts
WHERE birth_rate NOT NULL
ORDER BY birth_rate
LIMIT 20;

Done.


[('Monaco', 6.65),
 ('Saint Pierre and Miquelon', 7.42),
 ('Japan', 7.93),
 ('Andorra', 8.13),
 ('Korea, South', 8.19),
 ('Singapore', 8.27),
 ('Slovenia', 8.42),
 ('Germany', 8.47),
 ('Taiwan', 8.47),
 ('San Marino', 8.63),
 ('Greece', 8.66),
 ('Italy', 8.74),
 ('Bosnia and Herzegovina', 8.87),
 ('Macau', 8.88),
 ('Bulgaria', 8.92),
 ('Serbia', 9.08),
 ('Romania', 9.14),
 ('Hungary', 9.16),
 ('Hong Kong', 9.23),
 ('Portugal', 9.27)]

The countries and islands in Western Europe (Monaco, Andorra, Germany, Greece, Italy, Portugal), some small ones (Saint Pierre and Miquelon, Singapore, Taiwan, San Marino, Macao, Hong Kong), Japan, South Korea, and Eastern Europe have the lowest birth rates, which can reach 7 births per 1,000 people (Slovenia, Bosnia and Herzegovina, Bulgaria, Serbia, Romania, Hungary).

## Exploraing the Death rate

In [68]:
%%sql
SELECT name, death_rate AS top10_death_rate
  FROM facts
ORDER BY death_rate DESC
LIMIT 10;

Done.


[('Lesotho', 14.89),
 ('Ukraine', 14.46),
 ('Bulgaria', 14.44),
 ('Guinea-Bissau', 14.33),
 ('Latvia', 14.31),
 ('Chad', 14.28),
 ('Lithuania', 14.27),
 ('Namibia', 13.91),
 ('Afghanistan', 13.89),
 ('Central African Republic', 13.8)]

Up to 15 fatalities per 1,000 people are recorded in nations in Africa, Eastern Europe (Ukraine, Bulgaria, Serbia, Belarus, Hungary), the Baltic States, Afghanistan, and Russia. We can see that, despite the fact that African nations heavily dominate this list as well, all of them have significantly higher birth rates, which contributes to the highest rates of population growth. High death rates combined with low birth rates result in low population growth rates for the nations of Eastern Europe.

#### Next, we'll explore the BOTTOM20 countries by death rate.

In [70]:
%%sql
SELECT name, death_rate AS bottom10_death_rate
  FROM facts
WHERE death_rate NOT NULL
ORDER BY death_rate
LIMIT 10;

Done.


[('Qatar', 1.53),
 ('United Arab Emirates', 1.97),
 ('Kuwait', 2.18),
 ('Bahrain', 2.69),
 ('Gaza Strip', 3.04),
 ('Turks and Caicos Islands', 3.1),
 ('Saudi Arabia', 3.33),
 ('Oman', 3.36),
 ('Singapore', 3.43),
 ('West Bank', 3.5)]

The Middle East countries, historically known for having a very high standard of living, have the lowest death rates, up to 2 fatalities per 1,000 people.

Let's now look into the natural growth, or the gap between the birth and death rates, as another demographic measure.

In [72]:
%%sql
SELECT name, ROUND(birth_rate - death_rate, 1) AS top10_natural_increase
  FROM facts
WHERE death_rate NOT NULL AND birth_rate NOT NULL
ORDER BY top10_natural_increase DESC
LIMIT 10;

Done.


[('Malawi', 33.2),
 ('Uganda', 33.1),
 ('Niger', 33.0),
 ('Burundi', 32.7),
 ('Mali', 32.1),
 ('Burkina Faso', 30.3),
 ('Zambia', 29.5),
 ('Ethiopia', 29.1),
 ('South Sudan', 28.7),
 ('Tanzania', 28.4)]

According to population growth and birth rate trends, birth rates in African nations are significantly higher than mortality rates. Additionally, Iraq has had a significant natural rise.

In [74]:
%%sql
SELECT name, ROUND(birth_rate - death_rate, 1) AS bottom10_natural_increase
  FROM facts
WHERE death_rate NOT NULL AND birth_rate NOT NULL
ORDER BY bottom10_natural_increase
LIMIT 10;

Done.


[('Bulgaria', -5.5),
 ('Serbia', -4.6),
 ('Latvia', -4.3),
 ('Lithuania', -4.2),
 ('Ukraine', -3.7),
 ('Hungary', -3.6),
 ('Germany', -2.9),
 ('Slovenia', -2.9),
 ('Romania', -2.8),
 ('Belarus', -2.7)]

In [77]:
%%sql
SELECT name, migration_rate AS Top10_migration_rate
  FROM facts
WHERE migration_rate NOT NULL
ORDER BY migration_rate DESC
LIMIT 10;

Done.


[('Qatar', 22.39),
 ('American Samoa', 21.13),
 ('Micronesia, Federated States of', 20.93),
 ('Syria', 19.79),
 ('Tonga', 17.84),
 ('British Virgin Islands', 17.28),
 ('Luxembourg', 17.16),
 ('Cayman Islands', 14.4),
 ('Singapore', 14.05),
 ('Nauru', 13.63)]

In [78]:
%%sql
SELECT name, migration_rate AS Bottom10_migration_rate
  FROM facts
WHERE migration_rate NOT NULL
ORDER BY migration_rate ASC
LIMIT 10;

Done.


[('Andorra', 0.0),
 ('Argentina', 0.0),
 ('Azerbaijan', 0.0),
 ('Bahamas, The', 0.0),
 ('Belize', 0.0),
 ('Benin', 0.0),
 ('Bhutan', 0.0),
 ('Burkina Faso', 0.0),
 ('Burundi', 0.0),
 ('Central African Republic', 0.0)]

From the growth rate projection and mortality rate, we can say that Uganda, Malawi, and others will add most people to their populations next year.

It's interesting to note that eight countries will make up the majority of the increase in the global population over the next billion years, according to the <a href='https://www.livemint.com/news/world/india-pakistan-egypt-congo-ethiopia-nigeria-philippines-and-tanzania-likely-to-add-next-billion-as-world-population-touches-8-billion-11668511406123.html'> UN's</a> World Population Prospects 2022 report. This covers Tanzania, India, Pakistan, Ethiopia, the Congo, the Philippines, and Egypt.

Conclusion:    
- China, India, the United States, Indonesia, and Brazil have the largest populations. The Vatican and a few islands (Pitcairn, Niue, Tokelau, and the Cocos Islands) are the least populous.

Dynamics of the population

- The highest rates of births and deaths can be seen in African nations. The world's highest rates of natural increase and population expansion, however, are due to substantially higher birth rates.

- Monaco, Andorra, Germany, Greece, Italy, and Portugal are among the Western European nations with low birth rates. It results in a low population growth and a negative natural increase. The population growth for Germany and Italy, however, is not among the lowest ones because it appears that immigration has partially offset the negative natural increase in those countries.

- A negative natural increase is seen in many Baltic and Eastern European countries. It can be attributed to low birthrates in some of them (Bosnia and Herzegovina, Slovenia, Romania), high deathrates in others (Ukraine, Belarus, Baltic nations), and low birthrates and high deathrates in others (Bulgaria, Serbia, Hungary).

- The Middle East has the lowest global death rates across all of its nations.
- Iraq has a high rate of natural rise and population expansion since it is associated with nations with the highest birth rates.

- The birth and death rates in Afghanistan are among the highest in the world.

- The lowest birthrates in the world are found in South Korea, Saint Pierre and Miquelon, and Japan, which results in a negative natural increase.

- Russia has one of the highest death rates, which results in a low growth in the population and a negative natural increase.

- Area: Russia, Canada, the United States, China, and Brazil are the largest nations. Monaco and Vatican are the smallest.

- Water Area. When solely enclosed freshwater bodies of water are taken into account, the following nations stand out: Malawi (Lake Malawi), the Netherlands (artificial lakes Markermeer and IJsselmeer), and Uganda (Lake Victoria).